### DISCUSSION TITLE
Customizing the `handle` in `hooks.server.ts` with SvelteKit

### DISCUSSION THREAD

**Discussion by beeb** [nextauthjs/next-auth#6283] [Help] [answered]
As per the current [documentation](https://authjs.dev/reference/sveltekit/modules/main), one should export the `SvelteKitAuth` as the `handle` in `hooks.server.ts` for SvelteKit projects.

How can I further customize my `Handle` to add, for instance, localization information and so on? My current hook handle is not compatible with this usage and does the following:

- Check the URL to know if a language slug was passed
- Retrieve the user's preferred locale from the request headers
- Define some `locals` in the `event` passed to the `Handle`
- Return a `resolve` that replaces the `lang` property of the `html` tag with `transformPageChunk`

I cannot see how I could do this and use `@auth/sveltekit` at the same time.

Thanks

**Comment by ioucyf** [comment, marked as answer]
Have you tried the helper [sequence function](https://kit.svelte.dev/docs/modules#sveltejs-kit-hooks)?

It's like a function that allows you to run multiple handle function-like in specific sequential order; hence the name.

Example from the docs:

```js

import { sequence } from '@sveltejs/kit/hooks';
 
async function first({ event, resolve }) {
  console.log('first pre-processing');
  const result = await resolve(event, {
    transformPageChunk: ({ html }) => {
      // transforms are applied in reverse order
      console.log('first transform');
      return html;
    }
  });
  console.log('first post-processing');
  return result;
}
 
async function second({ event, resolve }) {
  console.log('second pre-processing');
  const result = await resolve(event, {
    transformPageChunk: ({ html }) => {
      console.log('second transform');
      return html;
    }
  });
  console.log('second post-processing');
  return result;
}
 
export const handle = sequence(first, second);

```

**Reply by beeb** [reply]
That's very useful! Sounds like it should fix my problem. When I have time to test it I'll report !

**Reply by beeb** [reply]
This might be a nice addition to the auth.js docs too, no time for a PR right this minute though!

**Reply by beeb** [reply]
Ok I tested and it works! Unlike in the example from the sveltekit docs, I didn't have to await and return the result, I simply returned the "resolve" without awaiting and it works like that too. Same goes for the `SvelteKitAuth()` which can be passed to `sequence` directly.

**Reply by ioucyf** [reply]
That's great! ^^

I'm happy to hear that I was able to help! Wish you the best. 

Thank you so much for the feedback; I appreciate it very much.
